src/fmeval/eval_algorithms/summarization_accuracy_semantic_robustness.py

```diff
@@ -178,129 +100,145 @@ class SummarizationAccuracySemanticRobustness(EvalAlgorithmInterface):
-                    use_stemmer_for_rouge=eval_algorithm_config.use_stemmer_for_rouge,
-                    model_type_for_bertscore=eval_algorithm_config.model_type_for_bertscore,
-                )
-            )
-            if summ_acc_actor is None
-            else summ_acc_actor
-        )
-
-    def __reduce__(self):  # pragma: no cover
-        """
-        Custom serializer method used by Ray when it serializes instances of this
-        class during dataset.map() operations.
-        """
-        serialized_data = (self._eval_algorithm_config, self._summarization_accuracy_eval_algo)
-        return SummarizationAccuracySemanticRobustness, serialized_data
-
-    def evaluate_sample(
+    def _build_pipeline(
         self,
-        model_input: str,
-        target_output: str,
         model: ModelRunner,
-        model_output: Optional[str] = None,
-        prompt_template: str = DEFAULT_PROMPT_TEMPLATE,
-    ) -> List[EvalScore]:  # type: ignore[override]
-        """
-        Summarization Accuracy Semantic Robustness evaluate sample.
-
-        :param model_input: text input for model
-        :param target_output: The expected responses from the model
-        :param model: An instance of ModelRunner which is the model under evaluation
-        :param model_output: The output of a model that we want to evaluate.
-        :param prompt_template: A template which can be used to compose prompt using model_input
-        :return: list of EvalScore object
+        prompt_template: str,
+        bertscore_model: Union[BertscoreHelperModel, ObjectRef],
+    ) -> TransformPipeline:
+        """Build the TransformPipeline to be used by `evaluate` and `evaluate_sample`.
+
+        While other evaluation algorithms (ex: Summarization Accuracy) can configure
+        their TransformPipeline at algorithm initialization, because the Summarization Accuracy
+        Semantic Robustness algorithm's evaluation logic depends on the ModelRunner
+        and prompt template that are evaluation-specific (i.e. these parameters aren't
+        configured at the algorithm level), the pipeline used by this algorithm is built
+        when `evaluate` or `evaluate_sample` is called.
+
+        :param model: The ModelRunner representing the model under evaluation.
+        :param prompt_template: A template that is used to construct the prompt fed to the model.
+        :param bertscore_model: Either a BertscoreHelperModel instance or a Ray actor handle corresponding
+            to a BertscoreHelperModel (i.e. a shared resource).
+        :returns: A TransformPipeline that can be used by either `evaluate_sample` or `evaluate`.
         """
-        util.require(
-            model_input,
-            "Missing required input: model_input, for SummarizationAccuracySemanticRobustness evaluate_sample",
-        )
-        util.require(
+        transforms = get_model_responses_from_perturbed_inputs(
```

Comment:
Ugh, I left this extraneous variable in from my old code. I will get rid of it in a followup PR, as we need to get the other PRs for the rest of the algos merged asap.